ring/ingester_lifecycle.go

```diff
@@ -16,125 +16,212 @@
 package ring
 
 import (
-	"encoding/json"
 	"fmt"
-	"hash/fnv"
 	"math/rand"
 	"net"
 	"os"
+	"sort"
 	"sync"
 	"time"
 
 	"github.com/prometheus/common/log"
 )
 
 const (
-	infName = "eth0"
+	infName           = "eth0"
+	ring              = "ring"
+	heartbeatInterval = 5 * time.Second
 )
 
 // IngesterRegistration manages the connection between the ingester and Consul.
 type IngesterRegistration struct {
-	consul ConsulClient
-	id     string
-	desc   []byte
+	consul    ConsulClient
+	numTokens int
 
-	quit chan struct{}
-	wait sync.WaitGroup
+	id       string
+	hostname string
+	quit     chan struct{}
+	wait     sync.WaitGroup
 }
 
 // RegisterIngester registers an ingester with Consul.
 func RegisterIngester(consulClient ConsulClient, listenPort, numTokens int) (*IngesterRegistration, error) {
-	desc, err := describeLocalIngester(listenPort, numTokens)
+	hostname, err := os.Hostname()
 	if err != nil {
 		return nil, err
 	}
-	buf, err := json.Marshal(desc)
+
+	addr, err := getFirstAddressOf(infName)
 	if err != nil {
 		return nil, err
 	}
 
 	r := &IngesterRegistration{
-		consul: consulClient,
-		id:     desc.ID,
-		desc:   buf,
-		quit:   make(chan struct{}),
+		consul:    consulClient,
+		numTokens: numTokens,
+
+		id:       hostname,
+		hostname: fmt.Sprintf("%s:%d", addr, listenPort),
```

Comment:
We advertise this in consul so the distributors know where to connect.  Will comment.